legend-depot-artifacts-services/src/main/java/org/finos/legend/depot/services/artifacts/purge/ArtifactsPurgeServiceImpl.java

```diff
@@ -120,6 +127,47 @@ public void delete(String groupId, String artifactId, String versionId)
         },decorateSpanWithVersionInfo(groupId, artifactId, versionId));
     }
 
+    @Override
+    public String deleteSnapshotVersions(String groupId, String artifactId, List<String> versions)
+    {
+        TracerFactory.get().executeWithTrace(DELETE_SNAPSHOT_VERSIONS,  () ->
+        {
+            Optional<StoreProjectData> storeProjectData = this.projects.findCoordinates(groupId, artifactId);
+            StoreProjectData project;
+            if (storeProjectData.isEmpty())
+            {
+                throw new IllegalArgumentException(String.format("No project found for %s-%s", groupId, artifactId));
+            }
+            else
+            {
+                project = storeProjectData.get();
+            }
+            String defaultBranch = VersionValidator.BRANCH_SNAPSHOT(project.getDefaultBranch() != null ? project.getDefaultBranch() : projectsConfiguration.getDefaultBranch());
+            List<String> nonCompliantVersionErrors = new ArrayList<>();
+            versions.forEach(versionId ->
+                    {
+                        if (!VersionValidator.isSnapshotVersion(versionId))
+                        {
+                            nonCompliantVersionErrors.add(String.format("Version %s can't be deleted as it is not a snapshot version", versionId));
+                        }
+                        else if (defaultBranch.equals(versionId))
+                        {
+                            nonCompliantVersionErrors.add(String.format("Version %s can't be deleted as it is the project's default branch", defaultBranch));
+                        }
+                        else
+                        {
+                            delete(groupId, artifactId, versionId);
+                        }
+                    });
+            if (!nonCompliantVersionErrors.isEmpty())
```

Comment:
for this one can we not throw an error but a proper response, also you return a null , so as a response will we always return null?